apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerCommandTextBuilder.cs

```diff
@@ -218,9 +220,9 @@ internal string DatabaseToDrop
         public static SpannerCommandTextBuilder FromCommandText(string commandText)
         {
             GaxPreconditions.CheckNotNullOrEmpty(commandText, nameof(commandText));
-            commandText = commandText.Trim();
+            var trimmedCommandText = RemoveCommentsAndStatementHint(commandText).Trim();
             // Split(new char[0]) splits the string using all whitespace characters.
-            var commandSections = commandText.Split((char[]) null, StringSplitOptions.RemoveEmptyEntries);
+            var commandSections = trimmedCommandText.Split((char[]) null, StringSplitOptions.RemoveEmptyEntries);
```

Comment:
In line 262 here (I can't leave a comment there), the code is doing `commandText = $"{commandName} {targetTable}";` and then a `SpannerCommandTextBuilder` is returned based on `commandText`, which means you loose any comments or hints, etc.

Reply:
That is true, but that is not a command that supports comments or hints anyways. That is a pseudo command text that indicates that the command should use a mutation and not a DML statement, and mutations do not support comments or hints.

Reply:
Good question. In this case the backend will not fail it, as the command text is only used to determine the type of command. The generated mutation will not contain the hint or comment, and will therefore be accepted by the backend. For comments, I would say that it's not really a problem. It basically means that we are introducing support for comments in these mutation command texts. For hints it might be better to fail in the client, as they will otherwise just silently be stripped away without any use.
Note: These mutation command texts is a .NET specific feature. Cloud Spanner (and also the client libraries for the other programming languages) do not support a command text like `INSERT <table_name>` or `UPDATE <table_name>`.

> Is EF Core generating mutations with comments?
No.